src/line.rs

```diff
@@ -0,0 +1,57 @@
+/// A macro for creating a [`Line`] using vec! syntax.
+///
+/// `line!` is similar to the [`vec!`] macro, but it returns a [`Line`] instead of a `Vec`.
+///
+/// # Examples
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::{line, raw};
+///
+/// let line = line!["hello", "world"];
+/// let line = line![raw!("hello {}", "world"), raw!("goodbye {}", "world")];
+/// let line = line!["hello".red(), "world".red().bold()];
+/// let line = line!["hello"; 2];
+/// ```
+///
+/// [`Line`]: crate::text::Line
+#[macro_export]
+macro_rules! line {
+    () => {
+        ratatui::text::Line::default()
+    };
+    ($span:expr; $n:expr) => {
+      ratatui::text::Line::from(vec![$span.into(); $n])
+    };
+    ($($span:expr),+ $(,)?) => {{
+        let mut line = ratatui::text::Line::default();
+        $(
+            line.push_span($span);
+         )+
+         line
```

Comment:
```suggestion
        ratatui::text::Line::from(vec![
        $(
            $span.into(),
        )+
        ])
```
Using `vec![]` rather than pushing into an existing `Vec` inside the Line results in less allocs as `vec![]` can internally use `Vec::with_capacity` as its length is known at compile size then. This should improve runtime performance.

(Wrote that macro in the browser, might not work)

Reply:
Nice!